source/LocationFilter.h

```diff
@@ -51,20 +53,26 @@ class LocationFilter {
 	bool IsEmpty() const;
 	bool IsValid() const;
 
+	// All public methods that take a PlayerInfo* can receive a nullptr, but
+	// that will disable all tests which require PlayerInfo. If a resulting
+	// LocationFilter has nothing else in it, it'll always match, which means
+	// a not filter of them will never match.
+
 	// If the player is in the given system, does this filter match?
-	bool Matches(const Planet *planet, const System *origin = nullptr) const;
-	bool Matches(const System *system, const System *origin = nullptr) const;
+	bool Matches(const Planet *planet, const PlayerInfo *player, const System *origin = nullptr) const;
+	bool Matches(const System *system, const PlayerInfo *player, const System *origin = nullptr) const;
```

Comment:
```suggestion
	bool Matches(const Planet *planet, const PlayerInfo *player = nullptr, const System *origin = nullptr) const;
	bool Matches(const System *system, const PlayerInfo *player = nullptr, const System *origin = nullptr) const;
```
and for all the others too please